dev/release/verify-release-candidate.sh

```diff
@@ -859,11 +859,12 @@ test_go() {
   maybe_setup_go || exit 1
   maybe_setup_conda compilers go=1.17 || exit 1
 
-  pushd go/arrow
+  pushd go
   go get -v ./...
-  go test ./...
+  go test ./...  
+  go install ./...  
   go clean -modcache
-  popd
+  popd  
```

Comment:
Looks there's a trailing space.

Do we want a `shellcheck` lint job in CI? No?